docs/ledger/migrations/v4.0-migration.md

```diff
@@ -0,0 +1,272 @@
+  - updated `batch_denom` to `denom`
+  - updated `metadata` from `bytes` to `string`
+
+- [v1alpha1.MsgCancel.CancelCredits](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.MsgCancel.CancelCredits) → [v1.Credits](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.Credits)
+
+  - scope and name change
+  - no changes to fields
+
+- [v1alpha1.ClassInfo](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.ClassInfo) → [v1.ClassInfo](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.ClassInfo)
+
+  - added `credit_type_abbrev`
+  - removed `issuers`
+  - removed `credit_type`
+  - removed `num_batches`
+  - updated `class_id` to `id`
+  - updated `metadata` from `bytes` to `string`
+
+- [v1alpha1.MsgCreateBatch.BatchIssuance](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.MsgCreateBatch.BatchIssuance) → [v1.BatchIssuance](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.BatchIssuance)
+
+  - scope change
+  - updated `retirement_location` to `retirement_jurisdiction`
+
+- [v1alpha1.MsgRetire.RetireCredits](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.MsgRetire.RetireCredits) → [v1.Credits](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.Credits)
+
+  - scope and name change
+  - no changes to fields
+
+- [v1alpha1.MsgSend.SendCredits](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.MsgSend.SendCredits) → [v1.MsgSend.SendCredits](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.MsgSend.SendCredits)
+
+  - updated `retirement_location` to `retirement_jurisdiction`
+
+### regen.ecocredit.basket
+
+Regen Ledger v4.0 includes non-breaking changes to [regen.basket.v1](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.basket.v1).
+
+#### Msg Service
+
+- [v1.MsgCreate](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.basket.v1#regen.ecocredit.basket.v1.MsgCreate)
+
+  - deprecated `exponent`
+  - updated `DateCriteria` (see types)
+
+- [v1.MsgCreateResponse](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.basket.v1#regen.ecocredit.basket.v1.MsgCreateResponse)
+
+  - no changes
+
+- [v1.MsgPut](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.basket.v1#regen.ecocredit.basket.v1.MsgPut)
+
+  - no changes
+
+- [v1.MsgPutResponse](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.basket.v1#regen.ecocredit.basket.v1.MsgPutResponse)
+
+  - no changes
+
+- [v1.MsgTake](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.basket.v1#regen.ecocredit.basket.v1.MsgTake)
+
+  - ***TODO: revert `retirement_jurisdiction` or add and deprecate `retirement_location`***
```

Comment:
is there an issue on regen-ledger for doing that?

Reply:
Not yet but I'll take care of this after morning meetings.

Reply:
https://github.com/regen-network/regen-ledger/pull/1192